community/index/src/main/java/org/neo4j/index/internal/gbptree/GBPTree.java

```diff
@@ -771,10 +762,10 @@ private void checkpoint( IOLimiter ioLimiter, Header.Writer headerWriter ) throw
 
         // Block writers, or if there's a current writer then wait for it to complete and then block
         // From this point and till the lock is released we know that the tree won't change.
-        long stamp = cleanerLock.writeLock();
-        writerLock.lock();
+        lock.writerAndCleanerLock();
         try
         {
+            assertRecoveryCleanSuccessful();
```

Comment:
Yes, it should be there because we need to check it after we take the lock.

We need to check for `null` because checkpoint is used during constructor when bumping generation before cleaner is initialised.